### PR TITLE
OMWorld: reenable all platforms

### PR DESCRIPTION
**Pull request by xmas92**
This reenables all platforms to use OMWorld, and by extension UseCompactObjectHeaders. 

This change simply calls the runtime if a lock is inflated, until port support for OMWorld cache lookup is added.

ARM (32-bit) required no changes as it already always called the runtime when a monitor is inflated.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [ ] Change must be properly reviewed (1 review required, with at least 1 [Committer](https://openjdk.org/bylaws#committer))

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/lilliput.git pull/174/head:pull/174` \
`$ git checkout pull/174`

Update a local copy of the PR: \
`$ git checkout pull/174` \
`$ git pull https://git.openjdk.org/lilliput.git pull/174/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 174`

View PR using the GUI difftool: \
`$ git pr show -t 174`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/lilliput/pull/174.diff">https://git.openjdk.org/lilliput/pull/174.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/lilliput/pull/174#issuecomment-2126333174)

### CODE REVIEW

**Review by coleenp — CHANGES_REQUESTED** [review]
This is sort of awkward because right now, Lightweight locking is default, so going forward, if we put the OMWorld table on a switch, we'll have to revert this code.  Would it be possible for this Lilliput patch to conditionalize the code on the UseCompactObjectHeaders instead?

**Comment by xmas92** [comment]
> This is sort of awkward because right now, Lightweight locking is default, so going forward, if we put the OMWorld table on a switch, we'll have to revert this code. Would it be possible for this Lilliput patch to conditionalize the code on the UseCompactObjectHeaders instead?

A UseCompactObjectHeaders condition will not work as you can still run LM_LIGHTWEIGHT with -UseCompactObjectHeaders. And that would crash. 

I was thinking this could be done when introducing the OMWorld flag. (And simply reintroduce the code this removes).

The alternative is that I wait with this patch until you have introduced the UseOMWorld flag. And I can then update this to instead of removing code condition the runtime call on the OMWorld flag.

File: src/hotspot/share/runtime/lightweightSynchronizer.cpp [hunk 1]
```diff
@@ -1042,6 +1042,16 @@ bool LightweightSynchronizer::quick_enter(oop obj, JavaThread* current, BasicLoc
 
   const markWord mark = obj->mark();
 
+  if (mark.is_unlocked()) {
```
Comment (rkennke) [inline src/hotspot/share/runtime/lightweightSynchronizer.cpp:1045]: Was that not needed before? Or why is this added now?

**Review by rkennke — COMMENTED** [review]
Hi Axel, this mostly looks good, I have just one question.

File: src/hotspot/share/runtime/lightweightSynchronizer.cpp (hunk 1 above)
Comment (coleenp) [reply src/hotspot/share/runtime/lightweightSynchronizer.cpp:1045]: Also wondering this.  We come here if the object is locked via slow path, but can it become unlocked only for the platforms that come don't check the ObjectMonitor and go slow path?

**Comment by coleenp** [comment]
The changes for this PR are also in this PR:  https://github.com/openjdk/lilliput/pull/181/   @rkennke  can you have a look at this? Thanks.

File: src/hotspot/share/runtime/lightweightSynchronizer.cpp (hunk 1 above)
Comment (xmas92) [reply src/hotspot/share/runtime/lightweightSynchronizer.cpp:1045]: While it is possible that someone has unlocked/deflated by the time we get here. This was really thought as a bandaid for `x86_32`.  Which now skips fast locking due to lack of a thread register in some paths. 

I think the whole `quick_enter` should be reevaluated in case when we get here from C2. @coleenp has been running some experiments where it is removed completely.

I can see a solution where C1 and C2 have two different entry points, where C1 does the `quick_enter` (and only fast_locking on `x86_32`) while the C2 entry point skips it. As it only gets here in a true slow path (except for `arm32`)

**Comment by xmas92** [comment]
Closing this as it was merged as a part of #181
